### PR TITLE
Making Enum conform to `Equatable` and `CaseIterable`

### PR DESCRIPTION
**Pull request by mackoj**
Remove old `cases` implementation and replacing it with `CaseIterable`

### CODE REVIEW

**Comment by yonaskolb** [comment]
Thanks @mackoj. Could you run the tests so that the fixtures get updated, and commit them. That's why the tests are failing.
Could you also add a changelog entry

**Comment by mackoj** [comment]
Ok, I will do it today.

**Comment by mackoj** [comment]
It actually breaks swift 4.1 :/

File: Templates/Swift/Includes/Enum.stencil [hunk 1]
```diff
@@ -1,14 +1,8 @@
 {% if description %}
 /** {{ description }} */
 {% endif %}
-public enum {{ enumName }}: {{ type }}, Codable {
+public enum {{ enumName }}: {{ type }}, Codable, Equatable, CaseIterable {
```
Comment (manojmahapatra) [inline Templates/Swift/Includes/Enum.stencil]: I think you can conditionally use the template for swift 4.2 Vs 4.1

```swift
#if swift(>=4.2)
// swift 4.2 goes here
#else 
// swift 4.1 goes here 
#endif
````

Comment (mackoj) [reply Templates/Swift/Includes/Enum.stencil]: Good idea I will do it.

Comment (mackoj) [reply Templates/Swift/Includes/Enum.stencil]: I have done it but I still have to fix the fixture...

**Comment by yonaskolb** [comment]
Perhaps a cleaner way would be to just define a custom `CaseIterable` below Swift 4.2, perhaps in `Coding.swift`
```swift
#if swift(<4.2)
public protocol CaseIterable {
    static let cases: [Self] { get }
}
#endif
```

**Comment by yonaskolb** [comment]
In the interest of time I've just opened another PR here https://github.com/yonaskolb/SwagGen/pull/133. I opted to simply drop Swift 4.1 support

**Pull request closed without merging**